### PR TITLE
generate version.h file per package using the ament_generate_version_header 

### PR DESCRIPTION
**Pull request by saikishor**
These changes of this PR allow the ros2_control packages to have the `version.h` generated by the  `ament_cmake_gen_version_h`

### CODE REVIEW

File: controller_interface/CMakeLists.txt [hunk 1]
```diff
@@ -90,3 +90,4 @@ install(TARGETS controller_interface
 ament_export_targets(export_controller_interface HAS_LIBRARY_TARGET)
 ament_export_dependencies(${THIS_PACKAGE_INCLUDE_DEPENDS})
 ament_package()
+ament_generate_version_header(controller_interface)
```
Comment (fmauch) [inline controller_interface/CMakeLists.txt:93]:
```suggestion
ament_generate_version_header(${PROJECT_NAME})
```

File: controller_manager/CMakeLists.txt [hunk 2]
```diff
@@ -226,3 +226,4 @@ ament_python_install_package(controller_manager
 ament_export_targets(export_controller_manager HAS_LIBRARY_TARGET)
 ament_export_dependencies(${THIS_PACKAGE_INCLUDE_DEPENDS})
 ament_package()
+ament_generate_version_header(controller_manager)
```
Comment (fmauch) [inline controller_manager/CMakeLists.txt:229]:
```suggestion
ament_generate_version_header(${PROJECT_NAME})
```

File: hardware_interface/CMakeLists.txt [hunk 3]
```diff
@@ -122,3 +122,4 @@ install(
 ament_export_targets(export_hardware_interface HAS_LIBRARY_TARGET)
 ament_export_dependencies(${THIS_PACKAGE_INCLUDE_DEPENDS})
 ament_package()
+ament_generate_version_header(hardware_interface)
```
Comment (fmauch) [inline hardware_interface/CMakeLists.txt:125]:
```suggestion
ament_generate_version_header(${PROJECT_NAME})
```

File: joint_limits/CMakeLists.txt [hunk 4]
```diff
@@ -61,3 +61,4 @@ install(TARGETS joint_limits
 ament_export_targets(export_joint_limits HAS_LIBRARY_TARGET)
 ament_export_dependencies(${THIS_PACKAGE_INCLUDE_DEPENDS})
 ament_package()
+ament_generate_version_header(joint_limits)
```
Comment (fmauch) [inline joint_limits/CMakeLists.txt:64]:
```suggestion
ament_generate_version_header(${PROJECT_NAME})
```

File: transmission_interface/CMakeLists.txt [hunk 5]
```diff
@@ -84,3 +84,4 @@ install(TARGETS transmission_interface
 ament_export_targets(export_transmission_interface HAS_LIBRARY_TARGET)
 ament_export_dependencies(${THIS_PACKAGE_INCLUDE_DEPENDS})
 ament_package()
+ament_generate_version_header(transmission_interface)
```
Comment (fmauch) [inline transmission_interface/CMakeLists.txt:87]:
```suggestion
ament_generate_version_header(${PROJECT_NAME})
```

**Review by fmauch — COMMENTED** [review]
Not mandatory, but AFAIK it is common practice to use `${PROJECT_NAME}` when possible.

**Review by bmagyar — APPROVED** [review]

**Comment by christophfroehlich** [comment]
@mergifyio backport humble

**Pull request merged**